### ISSUE TITLE
Iris doesn’t show cover art on some pages

### ISSUE THREAD

**Issue by tschofen** [jaedb/Iris#835]
### Iris version

3.62.0

### Operating system(s) affected

- [ ] Windows
- [ ] MacOS
- [ ] iOS
- [ ] Android
- [ ] Linux
- [X] Other

### Browser(s) affected

- [X] Firefox
- [X] Chrome
- [X] Edge
- [X] Other

### What happened?

The only page that consistently loads cover art, is the Albums page…which as far as I can see is loaded from the images extracted by Mopidy-Local.

I noticed that the Artists page doesn’t load any images until I look at an individual artist. It pulls that artist image from the web (e.g. discogs) and caches it. Going back to the artists page, does show the image loaded from discogs.

When I do a search, no images are displayed for artists or albums at all. However, visiting a specific Album or Artist will show the cover art and an artist thumbnail.
I was hoping to be able to see the Album cover art in the search results. Any idea what could cause this, and how I can fix it?

I’m running latest Mopidy, Mopidy-local and Mopidy-Iris on a RaspberryPi 4.

**Comment by jaedb** [comment]
Images that are not provided by the backend (ie Mopidy-Local, Mopidy-Ytmusic, Mopidy-Spotify, etc) will have artwork provided by Discogs (via Iris). To avoid hitting rate limits, artwork is not fetched en masse; it is only fetched when loading a specific asset (ie Artist, Album, Track, etc). This is why clicking into an Artist and then returning to Library -> Artists shows artwork for just that single artist.

Mopidy-Local does not provide Artist artwork, which is why they will always use Discogs and have the above behavior.

Digging deeper into your issue description, is the primary area where you are not seeing expected artwork (including Album art for local albums you have local artwork for) the Search view?

**Comment by tschofen** [comment]
Yes, that is correct. I see no album artwork when searching, even though album artwork does show up when looking at albums collection or individual albums.

Here's an example of the rendered HTML. Looks like the the URL of the background image is null:

`<a class=" " href="/iris/album/bG9jYWw6YWxidW06bWQ1OjNkNTRmYmYwZTM3ZjJmOWNhYjc4MGZjOWQ2ODUzMjdk/John Luther Adams%3A The Far Country">
    <div class="thumbnail thumbnail--loaded thumbnail--medium grid__item__thumbnail"><i class="icon icon--material thumbnail__placeholder">album</i>
        <div class="thumbnail__image" style="background-image: url(&quot;null&quot;);"></div>
        <div class="thumbnail__actions"></div>
    </div>
    <div class="grid__item__name">John Luther Adams: The Far Country</div>
    <div class="grid__item__secondary">
        <ul class="grid__item__secondary__content details"><span type="fontawesome" aria-hidden="true" class="fa fa-folder icon icon--fontawesome source"></span>
            <li><span class="undefined links-sentence"><span><span>Apollo Quartet</span></span></span></li>
        </ul>
    </div>
</a>`

**Comment by 9acca9** [comment]
@tschofen Im having the same trouble. Please, if you find a solution share it.
Thanks.

**Comment by tehkillerbee** [comment]
For Mopidy-Tidal, Iris seems to ignore the Artist images provided by Tidal and always falls back to discogs when clicking on the artist. Why does this happen?

**Comment by 1marc1** [comment]
I am having the same issue: artist image is loaded when you click on an artist. But when you close the web browser and open it again, the artist images are all gone again. For album images, they are are always there. Likely (in my case) this is because they may be embedded in the MP3s.

Is the idea that Iris stores the artist images and that something is broken on my system?
